src/plugins/mer/meremulatordevice.cpp

```diff
@@ -737,6 +819,29 @@ void MerEmulatorDeviceManager::onDeviceListReplaced()
             MerVirtualBoxManager::updateEmulatorQmlLivePorts(merEmulator->virtualMachine(),
                     merEmulator->qmlLivePortsList());
         m_deviceQmlLivePortsCache.insert(merEmulator->id(), nowQmlLivePorts);
+
+        const int nowMemorySize = merEmulator->memorySizeMb();
+        if (nowMemorySize != oldMemorySizeCache.value(merEmulator->id()))
+            MerVirtualBoxManager::setMemorySizeMb(merEmulator->virtualMachine(), nowMemorySize);
+
+        m_deviceMemorySizeCache.insert(merEmulator->id(), nowMemorySize);
+
+        const int nowCpuCount = merEmulator->cpuCount();
+        if (nowCpuCount != oldCpuCountCache.value(merEmulator->id()))
+            MerVirtualBoxManager::setCpuCount(merEmulator->virtualMachine(), nowCpuCount);
+
+        m_deviceCpuCountCache.insert(merEmulator->id(), nowCpuCount);
+
+        const int nowVdiCapacityMb = merEmulator->vdiCapacityMb();
+        if (nowVdiCapacityMb != oldVdiInfoCache.value(merEmulator->id())) {
+            QEventLoop loop;
+            MerVirtualBoxManager::setVdiCapacityMb(merEmulator->virtualMachine(), nowVdiCapacityMb, &loop, [&loop] (bool ok) {
+                Q_UNUSED(ok);
```

Comment:
Now I noticed for build engine we issue the "Some changes could not be saved!" message in case of errors (`MerOptionsWidget::store()`). We should add the same error handling here. Please use a separate commit for extending the original code, ordered before this commit.

Reply:
Add handling error